pages/api/users/[uuid].ts

```diff
@@ -65,7 +65,7 @@ async function handlePATCH({ req, res, queryUUID }: HandlerParams<User>) {
       uuid: queryUUID,
     },
     data: {
-      email: newUserDetails.email,
+      email: newUserDetails.email.toLowerCase(),
```

Comment:
Tässä näkee hyvin miksi `as` on vaarallinen. Koska kyseessä on PATCH endpoint kaikki kentät eivät ole pakollisia -> `newUserDetails.email` voi olla `undefined` jolloin pitää käyttää conditional chaining operaattoria. TypeScript ei osaa varoittaa, koska `const newUserDetails = req.body as User;` antaa sille virheellisen tyypin. Itse asiassa tuo tyyppi kannattaa muuttaa oikeaksi:
```ts
const newUserDetails = req.body as Partial<User>;
```


```suggestion
      email: newUserDetails.email?.toLowerCase(),
```